packages/firestore/test/unit/local/mutation_queue.test.ts

```diff
@@ -61,13 +57,14 @@ describe('IndexedDbMutationQueue', () => {
     return;
   }
 
-  beforeEach(() => {
-    return persistenceHelpers.testIndexedDbPersistence().then(p => {
-      persistence = p;
-    });
+  let persistencePromise: Promise<Persistence<IndexedDbTransaction>>;
+  let persistence: Persistence<IndexedDbTransaction>;
+  beforeEach(async () => {
+    persistencePromise = persistenceHelpers.testIndexedDbPersistence();
+    persistence = await persistencePromise;
   });
 
-  genericMutationQueueTests();
+  genericMutationQueueTests(() => persistencePromise);
```

Comment:
Is it possible to pass `persistence` here? Then you don't have to deal with `persistencePromise` anywhere. Alternatively, you could do:

`return persistencePromise.then(p => genericMutationQueueTests(p))`

Reply:
This unfortunately does not work, as it will break if another test uses `only`. The reason is that `describe` will always run, creating the persistence layer, but the `beforeEach` and `afterEach` methods will not, so it won't be properly cleaned up. The key is to only actually create the persistence layer in a `beforeEach` method.

Reply:
Actually, I believe this will break even without another test using `only`. At the time that you would return this, `persistencePromise` is `undefined`. `beforeEach` needs to run before you can use `persistencePromise`.

Reply:
Is this refactoring related to the TransactionType generic?  If so, is it obsolete now?  It seems fine, but just wanting to make sure it's not just leftover churn.

Reply:
I think leftover churn. I'm removing it.